Documentation/Books/Manual/Upgrading/VersionSpecific/Upgrading34.md

```diff
@@ -1,6 +1,32 @@
 Upgrading to ArangoDB 3.4
 =========================
 
-Please be sure that you have checked the list of [incompatible changes in 3.4](../../ReleaseNotes/UpgradingChanges34.md)
+Please be sure that you have checked the list of
+[incompatible changes in 3.4](../../ReleaseNotes/UpgradingChanges34.md)
 before upgrading.
 
+Upon upgrading from 3.3 to 3.4, the following storage engine-specific data conversion tasks
```

Comment:
But it's not fully correct. You could install a fresh 3.4 on a new system, but copy a 3.3 data folder from another machine. Then the conversion tasks need to be run as well. I don't see any direct relation between upgrading the binaries and the need to convert 3.3 -> 3.4 data. It is a common case, that you upgrade in-place and you have data that needs to be converted, but it's not the only circumstance a conversion is required.

Reply:
if you instal a fresh 3.4 and copy a 3.3 data folder and want to use this data folder with 3.4 you have to perform - as far as i know - an upgrade of the data directory, first, otherwise it is not expected to work.  I do not believe the conversion tasks will run, unless you perform the upgrade - which I believe is triggered if (in this manual case) you use the option --database.auto-upgrade .  Other "automatic" cases do that by their own.  We might want to document again this approach, as it was in 3.3 but we removed in 3.4 during the ugprade refactor

Reply:
simran, yes we should document this case here imho: https://docs.arangodb.com/3.4/Manual/Upgrading/Manually/ in a new page (before active failover and cluster) - but this is another task .  could you open issue in planning, please? P1? thanks

Reply:
> if you install a fresh 3.4 and copy a 3.3 data folder and want to use this data folder with 3.4 you have to perform - as far as i know - an upgrade of the data directory

That is also my understanding. If the datafiles are in the 3.3 format, then they need to be upgraded to the 3.4 format. This can occur if you do an in-place upgrade of the binaries but also if you try to start 3.4 arangod binary using a 3.3 data folder.

Datafile upgrades require the setting `database.auto-upgrade` to be set to `true`, but the [default is `false`](https://docs.arangodb.com/3.4/Manual/Programs/Arangod/Options.html#database-options). 

If you upgrade using the starter, it seems like you also have to [set this option manually](https://docs.arangodb.com/3.4/Manual/Upgrading/Starter/#deployment-mode-single). I'm not sure if there are any other ways like upgrading with an Installer (e.g. Windows) or script provided by us, which set this option automatically.

Reply:
Created a ticket and added it to the board: https://github.com/arangodb/planning/issues/3286